example/lib/onegini_listener.dart

```diff
@@ -161,22 +147,66 @@ class OneginiListener extends OneginiEventListener {
   }
 
   @override
-  void openCustomTwoStepRegistrationScreen(
+  void eventInitCustomRegistration(
+    BuildContext buildContext, String data) {
+
+    try {
+      var response = jsonDecode(data);
+      var providerId = response["providerId"];
+
+      if (providerId == "2-way-otp-api") {
+        // a 2-way-otp does not require data for the initialization request
+        OneginiCustomRegistrationCallback()
+          .submitSuccessAction(providerId, null)
+          .catchError((error) => {
+            if (error is PlatformException) {
+              showErrorToast(error.message)
+            }
+          });
+      }
+    } on FormatException catch (error) {
+      showErrorToast(error.message);
+      return;
+    }
+  }
+
+  @override
+  void eventFinishCustomRegistration(
       BuildContext buildContext, String data) {
-    var response = jsonDecode(data);
-    if (response["providerId"] == "2-way-otp-api")
-      Navigator.push(
-        buildContext,
-        MaterialPageRoute(
-            builder: (context) => OtpScreen(
-                  password: response["data"],
-                )),
-      );
+
+    try {
+      var response = jsonDecode(data);
+      var providerId = response["providerId"];
+
+      if (providerId == "2-way-otp-api")
+        Navigator.push(
+          buildContext,
+          MaterialPageRoute(
+              builder: (context) => OtpScreen(
+                    password: response["data"],
+                    providerId: providerId
+                  )),
+        );
+    } on FormatException catch (error) {
+      showErrorToast(error.message);
+      return;
+    }
   }
 
   @override
   void handleRegisteredUrl(BuildContext buildContext, String url) async {
     await Onegini.instance.userClient.handleRegisteredUserUrl(buildContext, url,
         signInType: WebSignInType.insideApp);
   }
+
+  void showErrorToast(String errorMessage) {
```

Comment:
You should move this generic error toast to separate file and use just this one from multiple places instead of declaring same method in few files.

Reply:
Created new class for it and used it on the relevant places of this PR. Created ticket https://onewelcome.atlassian.net/browse/FP-39 to add it on the remaining places